packages/astro/src/default/components/Logo.astro

```diff
@@ -0,0 +1,43 @@
+---
+import fs from 'node:fs';
+import path from 'node:path';
+
+const LOGO_EXTENSIONS = ['svg', 'png', 'jpeg', 'jpg'];
+
+interface Props {
+  logoLink: string;
+}
+
+function readLogoFile(logoPrefix: string) {
+  let logo;
+
+  for (const logoExt of LOGO_EXTENSIONS) {
+    const logoFilename = `${logoPrefix}.${logoExt}`;
+    const exists = fs.existsSync(path.join('public', logoFilename));
+
+    if (exists) {
+      logo = `/${logoFilename}`;
+      break;
+    }
+  }
+
+  return logo;
+}
+
+const { logoLink } = Astro.props;
+
+const logo = readLogoFile('logo');
+const logoDark = readLogoFile('logo-dark') ?? logo;
+
+if (!logo) {
+  console.warn(`No logo found in public/. Supported filenames are: logo.(${LOGO_EXTENSIONS.join('|')})`);
```

Comment:
Should we also mention that we look for a `logo-dark`?

Reply:
I was thinking of leaving it to the docs given that it's optional. Or maybe we could say something like:

```js
`
No logo found in public/. 
Supported filenames are: logo.(${LOGO_EXTENSIONS.join('|')})
(A logo-dark.(${LOGO_EXTENSIONS.join('|')}) will be used if found for the dark theme)`
```

Wdyt?

Reply:
I've added a mention for `logo-dark`. :+1: